cmd/settlement/bitflyer.go

```diff
@@ -0,0 +1,309 @@
+
+	// add this command as a settlement subcommand
+	SettlementCmd.AddCommand(BitflyerSettlementCmd)
+
+	// setup the flags
+	tokenBuilder := cmd.NewFlagBuilder(GetBitflyerTokenCmd)
+	uploadCheckStatusBuilder := cmd.NewFlagBuilder(UploadBitflyerSettlementCmd).
+		AddCommand(CheckStatusBitflyerSettlementCmd)
+	allBuilder := tokenBuilder.Concat(uploadCheckStatusBuilder)
+
+	uploadCheckStatusBuilder.Flag().String("input", "",
+		"the file or comma delimited list of files that should be utilized. both referrals and contributions should be done in one command in order to group the transactions appropriately").
+		Require().
+		Bind("input")
+
+	uploadCheckStatusBuilder.Flag().String("out", "./bitflyer-settlement",
+		"the location of the file").
+		Bind("out").
+		Env("OUT")
+
+	uploadCheckStatusBuilder.Flag().String("bitflyer-source-from", "self",
+		"tells bitflyer where to draw funds from").
+		Bind("bitflyer-source-from").
+		Env("BITFLYER_SOURCE_FROM")
+
+	uploadCheckStatusBuilder.Flag().String("bitflyer-dryrun", "",
+		"tells bitflyer that this is a practice round").
+		Bind("bitflyer-dryrun").
+		Env("BITFLYER_DRYRUN")
+
+	uploadCheckStatusBuilder.Flag().String("bitflyer-client-token", "",
+		"the token to be sent for auth on bitflyer").
+		Bind("bitflyer-client-token").
+		Env("BITFLYER_CLIENT_TOKEN")
+
+	tokenBuilder.Flag().String("bitflyer-client-id", "",
+		"tells bitflyer what the client id is during token generation").
+		Bind("bitflyer-client-id").
+		Env("BITFLYER_CLIENT_ID")
+
+	tokenBuilder.Flag().String("bitflyer-client-secret", "",
+		"tells bitflyer what the client secret during token generation").
+		Bind("bitflyer-client-secret").
+		Env("BITFLYER_CLIENT_SECRET")
+
+	tokenBuilder.Flag().String("bitflyer-extra-client-secret", "",
+		"tells bitflyer what the extra client secret is during token generation").
+		Bind("bitflyer-extra-client-secret").
+		Env("BITFLYER_EXTRA_CLIENT_SECRET")
+
+	allBuilder.Flag().String("bitflyer-server", "",
+		"the bitflyer domain to interact with").
+		Bind("bitflyer-server").
+		Env("BITFLYER_SERVER")
+}
+
+// BitflyerUploadSettlement marks the settlement file as complete
+func BitflyerUploadSettlement(
+	ctx context.Context,
+	action, inPath, outPath, token, sourceFrom string,
+	dryRun *bitflyer.DryRunOption,
+) error {
+	logger, err := appctx.GetLogger(ctx)
+	if err != nil {
+		_, logger = logging.SetupLogger(ctx)
+	}
+
+	// logger.Info().
+	// 	Str("action", action).
+	// 	Str("inPath", inPath).
+	// 	Str("outPath", outPath).
+	// 	Str("token", token).
+	// 	Str("sourceFrom", sourceFrom).
+	// 	Bool("dryRun", dryRun).
+	// 	Msg("stop")
+	// return nil
+	bulkPayoutFiles := strings.Split(inPath, ",")
+	bitflyerClient, err := bitflyer.New()
+	if err != nil {
+		logger.Error().Err(err).Msg("failed to create new bitflyer client")
+		return err
+	}
+	// set the auth token
+	if token == "" {
+		return errors.New("a token must be set at BITFLYER_CLIENT_TOKEN")
+	}
+	bitflyerClient.SetAuthToken(token)
```

Comment:
I think it would be cleaner overall if we had the client just do the auth token setting automatically, instead of having the user specify the auth token.  As I understand the process, step one is run the command to get the token from client credentials.  Step two is set that token in an environment variable.  Step three is run the actual command you want to run.

Can we just have the client perform the token retrieval so we can remove step 1 and 2 if the AuthToken on the client is unset?

Reply:
a bit difficult to embed into the module itself, but should be easy enough to do so on the application level

Reply:
:+1: